File: backend-node/routes/export.js
const express = require('express');
const path    = require('path');
const fs      = require('fs');
const { exportExcel, exportPDF, exportWord } = require('../services/exportService');

const router = express.Router();
const UPLOADS_DIR = process.env.UPLOADS_DIR || path.join(__dirname, '..', 'uploads');

function loadSessionData(sessionId) {
  const dir      = path.join(UPLOADS_DIR, sessionId);
  const stmtPath = path.join(dir, 'statements.json');
  if (!fs.existsSync(stmtPath)) return null;
  return JSON.parse(fs.readFileSync(stmtPath, 'utf8'));
}

// POST /export/excel
router.post('/excel', async (req, res) => {
  try {
    const { session_id } = req.body;
    if (!session_id) return res.status(400).json({ detail: 'session_id required' });

    const data = loadSessionData(session_id);
    if (!data) return res.status(404).json({ detail: 'No statements found. Please generate statements first.' });

    const buf = await exportExcel(data.statements, data.notes, data.meta);
    const filename = `${(data.meta?.entity_name || 'FinancialStatements').replace(/[^a-z0-9]/gi,'_')}_${data.meta?.period_end || 'report'}.xlsx`;

    res.setHeader('Content-Type', 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet');
    res.setHeader('Content-Disposition', `attachment; filename="${filename}"`);
    res.send(buf);
  } catch (err) {
    console.error('[export/excel]', err);
    res.status(500).json({ detail: err.message });
  }
});

// POST /export/pdf
router.post('/pdf', async (req, res) => {
  try {
    const { session_id } = req.body;
    if (!session_id) return res.status(400).json({ detail: 'session_id required' });

    const data = loadSessionData(session_id);
    if (!data) return res.status(404).json({ detail: 'No statements found. Please generate statements first.' });

    const buf = await exportPDF(data.statements, data.notes, data.meta);
    const filename = `${(data.meta?.entity_name || 'FinancialStatements').replace(/[^a-z0-9]/gi,'_')}_${data.meta?.period_end || 'report'}.pdf`;

    res.setHeader('Content-Type', 'application/pdf');
    res.setHeader('Content-Disposition', `attachment; filename="${filename}"`);
    res.send(buf);
  } catch (err) {
    console.error('[export/pdf]', err);
    res.status(500).json({ detail: err.message });
  }
});

// POST /export/word
router.post('/word', async (req, res) => {
  try {
    const { session_id } = req.body;
    if (!session_id) return res.status(400).json({ detail: 'session_id required' });

    const data = loadSessionData(session_id);
    if (!data) return res.status(404).json({ detail: 'No statements found. Please generate statements first.' });

    const buf = await exportWord(data.statements, data.notes, data.meta);
    const filename = `${(data.meta?.entity_name || 'FinancialStatements').replace(/[^a-z0-9]/gi,'_')}_${data.meta?.period_end || 'report'}.docx`;

    res.setHeader('Content-Type', 'application/vnd.openxmlformats-officedocument.wordprocessingml.document');
    res.setHeader('Content-Disposition', `attachment; filename="${filename}"`);
    res.send(buf);
  } catch (err) {
    console.error('[export/word]', err);
    res.status(500).json({ detail: err.message });
  }
});

module.exports = router;

File: backend/routers/export.py
import json
from pathlib import Path
from fastapi import APIRouter, HTTPException
from fastapi.responses import Response
from models.schemas import ExportRequest
from services.export_service import export_excel, export_pdf, export_word

router = APIRouter(prefix="/export", tags=["export"])
import os
if os.environ.get("VERCEL"):
    UPLOAD_DIR = Path("/tmp") / "uploads"
else:
    UPLOAD_DIR = Path(__file__).parent.parent / "uploads"
UPLOAD_DIR.mkdir(exist_ok=True)


def _load(session_id: str):
    session_dir = UPLOAD_DIR / session_id
    sf = session_dir / "statements.json"
    nf = session_dir / "notes.json"
    if not sf.exists():
        raise HTTPException(404, "Statements not found. Generate them first.")
    with open(sf) as f:
        statements = json.load(f)
    notes = []
    if nf.exists():
        with open(nf) as f:
            notes = json.load(f)
    return statements, notes


@router.post("/excel")
def export_to_excel(req: ExportRequest):
    statements, notes = _load(req.session_id)
    statements.update({"entity_name": req.entity_name, "period_end": req.period_end})
    payload = export_excel(statements, notes if req.include_notes else [], req.currency)
    filename = f"{req.entity_name.replace(' ', '_')}_FinancialStatements.xlsx"
    return Response(
        content=payload,
        media_type="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
        headers={"Content-Disposition": f'attachment; filename="{filename}"'},
    )


@router.post("/pdf")
def export_to_pdf(req: ExportRequest):
    statements, notes = _load(req.session_id)
    statements.update({"entity_name": req.entity_name, "period_end": req.period_end})
    payload = export_pdf(statements, notes if req.include_notes else [], req.currency)
    filename = f"{req.entity_name.replace(' ', '_')}_FinancialStatements.pdf"
    return Response(
        content=payload,
        media_type="application/pdf",
        headers={"Content-Disposition": f'attachment; filename="{filename}"'},
    )


@router.post("/word")
def export_to_word(req: ExportRequest):
    statements, notes = _load(req.session_id)
    statements.update({"entity_name": req.entity_name, "period_end": req.period_end})
    payload = export_word(statements, notes if req.include_notes else [], req.currency)
    filename = f"{req.entity_name.replace(' ', '_')}_FinancialStatements.docx"
    return Response(
        content=payload,
        media_type="application/vnd.openxmlformats-officedocument.wordprocessingml.document",
        headers={"Content-Disposition": f'attachment; filename="{filename}"'},
    )
